doc/specs/stdlib_linalg.md

```diff
@@ -206,3 +206,107 @@ program demo_outer_product
     !A = reshape([3., 6., 9., 4., 8., 12.], [3,2])
 end program demo_outer_product
 ```
+
+## `zeros/ones` - creates an vector or matrix of `integer` type, given shape and `0/1` value.
+
+### Status
+
+Experimental
+
+### Class
+
+Pure function.
+
+### Description
+
+`zeros/ones` creates an vector or matrix of `integer` type, given shape and `0/1` value.
+
+### Syntax
+
+For vector:
+`result = [[stdlib_linalg(module):zeros(interface)]](dim)`  
+`result = [[stdlib_linalg(module):ones(interface)]](dim)`
+
+For matrix:
+`result = [[stdlib_linalg(module):zeros(interface)]](dim1, dim2)`  
+`result = [[stdlib_linalg(module):ones(interface)]](dim1, dim2)`
+
+
+### Arguments
+
+`dim/dim1`: Shall be an `integer` type.
+
+`dim2`: Shall be an `integer` type.
+
+### Return value
+
+Returns an vector or matrix of `integer` type, given shape and `0/1` value.
+
+### Example
+
+```fortran
+program demo
+    use stdlib_linalg, only: zeros, ones
+    implicit none
+    real, allocatable :: A(:,:)
+    
+    A = zeros(2,2)          !! 0 0 0 0
+    A = ones(4,4)           !! 1 1 1 1; 1 1 1 1; 1 1 1 1; 1 1 1 1
+    A = 2.0*ones(2,2)       !! 2.00000000       2.00000000       2.00000000       2.00000000
+
+end program demo
+```
+
+## `ex` - creates an vector or matrix of `integer/logical/real/complex/string_type` type, given shape and `value` value.
```

Comment:
I think we have an intrinsic which can deal with this already: [spread](https://gcc.gnu.org/onlinedocs/gfortran/SPREAD.html)

Reply:
My intention is that `zeros` and `ones` cannot directly create the following statement based on exist `Fortran` snytax:
```fortran
real(kind=4, 8, 16) :: x = ones(2, 2)
```
That's why I want to create an `ex` function. I think it serves `stdlib_linalg`, which is simpler. There is indeed a little overlap with the `spread` function.
```Fortran
x = ex(1.0, 2,2) !! Same as 1.0*ones(2,2)
y = ex(1.0d0, 2) !! Same as spread(1.0d0, 1, 2) and 1.0*ones(2)
z = ex(1.0d0, 2,2) !! Not same as spread?
```

Reply:
The equivalent for `ex` for matrices is a combination of `spread` and `reshape`
```
z = reshape(spread(1.0d0, 2*2), [2, 2])
```

`ex` is rather short, how about `expand` instead.

Reply:
That is cool:
```fortran
z = reshape(spread(1.0d0, 2*2), [2, 2]) !! It is cool, but more complicated
z = expand(1.0d0, 2, 2) !! It is more simpler.
```
Although I always prefer shorter names, I like `expand`, thanks!